File: VB/Serialization/MyXmlXtraSerializer.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Utils
Imports DevExpress.Utils.Serializing
Imports DevExpress.Utils.Serializing.Helpers

Namespace Serialization
	Public Class MyXmlXtraSerializer
		Inherits XmlXtraSerializer
		Public Sub New()
			MyBase.New()
		End Sub

		Protected Overrides Function CreateSerializeHelper(ByVal rootObj As Object, ByVal useRootObj As Boolean) As SerializeHelper
			Return If(useRootObj, New MySerializeHelper(rootObj), New MySerializeHelper())
		End Function

		Protected Overrides Overloads Sub DeserializeObject(ByVal obj As Object, ByVal store As IXtraPropertyCollection, ByVal options As OptionsLayoutBase)
			If store Is Nothing Then
				Return
			End If

			If options Is Nothing Then
				options = OptionsLayoutBase.FullLayout
			End If

			Dim coll As New XtraPropertyCollection()
			coll.AddRange(store)

			Dim helper As DeserializeHelper = New MyDeserializeHelper(obj)
			helper.DeserializeObject(obj, coll, options)
		End Sub
	End Class
End Namespace

File: CS/Serialization/MyXmlXtraSerializer.cs
using DevExpress.Utils;
using DevExpress.Utils.Serializing;
using DevExpress.Utils.Serializing.Helpers;

namespace Serialization
{
	public class MyXmlXtraSerializer : XmlXtraSerializer
	{
		public MyXmlXtraSerializer()
			: base()
		{
		}

		protected override SerializeHelper CreateSerializeHelper(object rootObj, bool useRootObj)
		{
			return useRootObj ? new MySerializeHelper(rootObj) : new MySerializeHelper();
		}

		protected override void DeserializeObject(object obj, IXtraPropertyCollection store, OptionsLayoutBase options)
		{
			if ( store == null )
				return;

			if ( options == null )
				options = OptionsLayoutBase.FullLayout;
			
			XtraPropertyCollection coll = new XtraPropertyCollection();
			coll.AddRange(store);
			
			DeserializeHelper helper = new MyDeserializeHelper(obj);
			helper.DeserializeObject(obj, coll, options);
		}
	}
}
